### PR TITLE
doc: add some info about writing docs

### PR DESCRIPTION
**Pull request by emillon**

### CODE REVIEW

File: doc/hacking.rst [hunk 1]
```diff
@@ -331,6 +331,63 @@ For automatically updated builds, you can install sphinx-autobuild, and run
...
+.. _Title Case: https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case
+
+For project names, use the following capitalization:
+
+- **Dune** is the project, ``dune`` is the command. Files are called Dune files.
+- ``dune-project`` should always be written in monospace.
+- **OCaml**
+- **OCamlFormat**, and ``ocaml`` is the command.
+- ``odoc``, always in monospace.
+- **opam**. Can be capitalised as Opam. The command is ``opam``.
```
Comment (christinerose) [inline doc/hacking.rst:383]:
```suggestion
- **opam**. Can be capitalised as Opam in titles and at the beginning of sentences only, as the official name is formatted opam. The command is ``opam``.
```

**Review by christinerose — CHANGES_REQUESTED** [review]
The opam team has verified it should be lowercase like this unless in a title or beginning of sentence.

File: doc/hacking.rst [hunk 2]
```diff
@@ -331,6 +331,63 @@ For automatically updated builds, you can install sphinx-autobuild, and run
...
+Use American spelling.
+
+Use `Title Case`_ for titles and headings (every word except "little words"
+like of, and, or, etc.).
+
+.. _Title Case: https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case
+
+For project names, use the following capitalization:
+
+- **Dune** is the project, ``dune`` is the command. Files are called Dune files.
```
[the remark this answers is not in the document]
Comment (christinerose) [reply doc/hacking.rst:378]: Nice catch! I was told back when I started working on the Dune project, that `dune` files were in mono space as well as the command `dune`. Dune is the project and when talking about the Dune tool in general. :)

[the remark this answers is not in the document]
Comment (emillon) [reply doc/hacking.rst:378]: right, let's go with ``dune`` files.

**Pull request merged**